sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.Request.cs

```diff
@@ -41,403 +66,115 @@ public override string ClientRequestId
-                                }
-                            }
-                            break;
-                        case List<string> listValue:
-                            if (!currentRequest.Headers.TryAddWithoutValidation(headerName, listValue))
-                            {
-                                if (currentContent != null && !currentContent.Headers.TryAddWithoutValidation(headerName, listValue))
-                                {
-                                    throw new InvalidOperationException($"Unable to add header {headerName} to header collection.");
-                                }
-                            }
-                            break;
-                    }
-                }
+            #endregion
 
-                AddPropertiesForBlazor(currentRequest);
-
-                return currentRequest;
-            }
+            #region Overrides for "Core" methods from the PipelineRequest Template pattern
 
-            private static void AddPropertiesForBlazor(HttpRequestMessage currentRequest)
+            protected override string MethodCore
             {
-                // Disable response caching and enable streaming in Blazor apps
-                // see https://github.com/dotnet/aspnetcore/blob/3143d9550014006080bb0def5b5c96608b025a13/src/Components/WebAssembly/WebAssembly/src/Http/WebAssemblyHttpRequestMessageExtensions.cs
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))
+                get => _pipelineRequest.Method;
+                set
                 {
-                    SetPropertiesOrOptions(currentRequest, "WebAssemblyFetchOptions", new Dictionary<string, object> { { "cache", "no-store" } });
-                    SetPropertiesOrOptions(currentRequest, "WebAssemblyEnableStreamingResponse", true);
+                    // Update fields on both Request and PipelineRequest.
+                    base.MethodCore = value;
+                    _pipelineRequest.Method = value;
                 }
             }
 
-            private static string GetHttpHeaderValue(string headerName, object value) => value switch
+            protected override Uri? UriCore
             {
-                string headerValue => headerValue,
-                List<string> headerValues => string.Join(",", headerValues),
-                _ => throw new InvalidOperationException($"Unexpected type for header {headerName}: {value?.GetType()}")
-            };
-
-            public override void Dispose()
-            {
-                _headers.Dispose();
-                var content = Content;
-                if (content != null)
+                get
                 {
-                    Content = null;
-                    content.Dispose();
-                }
-            }
-
-            public override string ToString() => BuildRequestMessage(default).ToString();
+                    Uri uri = Uri.ToUri();
```

Comment:
Can we add a comment or two to make it clear that this line is about URIs?   Not sure if that's clear from context.

_(this is not a serious suggestion.  😜 )_